packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -340,14 +342,15 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
 
               case 'tool-call-end': {
                 // Post the full tool call now that we have all of the arguments.
-
                 controller.enqueue({
                   type: 'tool-call',
                   toolCallId: pendingToolCallDelta.toolCallId,
                   toolName: pendingToolCallDelta.toolName,
                   toolCallType: 'function',
                   args: JSON.stringify(
-                    JSON.parse(pendingToolCallDelta.argsTextDelta),
+                    JSON.parse(
+                      pendingToolCallDelta.argsTextDelta?.trim() || '{}',
```

Comment:
`??`

Reply:
This was intentional as I wanted the empty string to also turn into `{}` to parse successfully

Reply:
might be worth a comment